DXRModules/DeusEx/Classes/DXRBingoCampaign.uc

```diff
@@ -0,0 +1,391 @@
+            break;
+        case "08_NYC_STREET":
+            GetConversation('M08JockExit').AddFlagRef(BingoFlagM08, true);
+            break;
+        case "09_NYC_GRAVEYARD":
+            GetConversation('M09JockLeave').AddFlagRef(BingoFlagM09, true);
+            break;
+        case "10_PARIS_CHATEAU":
+            UpdateCryptDoors();
+            break;
+        case "11_PARIS_EVERETT":
+            GetConversation('TakeOff').AddFlagRef(BingoFlagM11, true);
+            break;
+        case "12_Vandenberg_GAS":
+            GetConversation('M12JockFinal').AddFlagRef(BingoFlagM12, true);
+            GetConversation('M12JockFinal2').AddFlagRef(BingoFlagM12, true);
+            break;
+        case "14_OCEANLAB_SILO":
+            GetConversation('JockArea51').AddFlagRef(BingoFlagM14, true);
+            break;
+    }
+}
+
+function NewBingoBoard()
+{
+    local DXREvents events;
+
+    foreach AllActors(class'DXREvents', events) break;
+    if (events == None) return;
+
+    dxr.flags.settings.starting_map = dxr.dxInfo.missionNumber * 10;
+    dxr.flags.bingoBoardRoll = 0;
+    dxr.flags.bingo_duration = 1;
+    events.CreateBingoBoard();
+    ClearDataVaultImages(player());
+}
+
+function AddBingoEventBlocker(name blockedTag, name bingoFlag) {
+    local #var(prefix)FlagTrigger ft;
+    local Actor blocked;
+    local name glue;
+
+    glue = dxr.flagbase.StringToName(blockedTag $ "_bingoblocked");
+
+    foreach AllActors(class'Actor', blocked, blockedTag) {
+        if (
+            MapExit(blocked) != None ||
+            InterpolateTrigger(blocked) != None ||
+            DataLinkTrigger(blocked) != None ||
+            Dispatcher(blocked) != None ||
+            DeusExMover(blocked) != None
+        ) {
+            blocked.tag = glue;
+        }
+    }
+
+    ft = Spawn(class'#var(prefix)FlagTrigger',, blockedTag);
+    ft.event = glue;
+    ft.flagName = bingoFlag;
+    ft.bTrigger = true;
+    ft.bTriggerOnceOnly = false;
+    ft.bSetFlag = false;
+    ft.SetCollision(false, false, false);
+}
+
+function UpdateCryptDoors()
+{
+    local #var(prefix)FlagTrigger ft;
+
+    if (
+        dxr.localURL == "10_PARIS_CHATEAU" &&
+        dxr.flagbase.GetBool('everettsignal') &&
+        dxr.flagbase.GetBool(BingoFlagM10) &&
+        IsBingoEnd(dxr.dxInfo.missionNumber, dxr.flags.bingo_duration) &&
+        !dxr.flagbase.GetBool('DXRando_CommandosUnhidden')
+    ) {
+        foreach AllActors(class'#var(prefix)FlagTrigger', ft, 'everettsignaldoor') {
+            ft.Trigger(None, None);
+            break;
+        }
+    }
+}
+
+static function bool IsBingoEnd(int missionNumber, int bingo_duration)
+{
+    if (missionNumber > 12) {
+        missionNumber -= 2;
+    } else if (missionNumber > 6) {
+        missionNumber -= 1;
+    }
+    return missionNumber % bingo_duration == 0 || missionNumber == 13;
+}
+
+static function name GetBingoMissionFlag(int missionNumber) {
```

Comment:
you can just do

```

StringToName("DXRando_Mission" $ missionNumber $ "_BingoCompleted")

```

Reply:
As I said in another comment, I want the constants defined at the top to be the authority on the flag names.